### PR TITLE
SONAR-7635 WS returns english localization of metric names and descriptions

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/metric/ws/MetricConverter.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+  private static class MetricDtoToWsMetric implements Function<MetricDto, Metric> {
+    private final I18n i18n;
+
+    private MetricDtoToWsMetric(I18n i18n) {
+      this.i18n = i18n;
+    }
+
+    @Override
+    public Metric apply(@Nonnull MetricDto dto) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/metric/ws/MetricConverter.java]: MAJOR Remove this "Nonnull" annotation to honor the overridden method's contract. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2638)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: sonar-core/src/main/resources/org/sonar/l10n/core.properties [hunk 2]
```diff
@@ -2266,17 +2276,31 @@ metric.generated_ncloc.name.suffix=generated lines of code
 metric.generated_ncloc.description=Generated non Commenting Lines of Code
 metric.generated_ncloc.abbreviation=Gen. LOC
 
+metric.last_commit_date.name=Date of Last Commit
+
```
Comment (julienlancelot) [inline sonar-core/src/main/resources/org/sonar/l10n/core.properties]: Description of last_commit_date is missing

File: sonar-core/src/main/resources/org/sonar/l10n/core.properties [hunk 3]
```diff
@@ -2732,12 +2768,6 @@ metric.profile_version.name=Profile Version
 metric.profile_version.description=Selected quality profile version
 
 
-#--------------------------------------------------------------------------------------------------------------------
-#
-# ISSUES METRICS
-#
-#--------------------------------------------------------------------------------------------------------------------
-
```
Comment (julienlancelot) [inline sonar-core/src/main/resources/org/sonar/l10n/core.properties]: Why removing this ?

File: sonar-core/src/test/java/org/sonar/core/i18n/DefaultI18nTest.java [hunk 4]
```diff
@@ -43,37 +46,62 @@
...
+    assertThat(underTest.getPropertyKeys().contains("bla_bla_bla")).isFalse();
+  }
+
+  @Test
+  public void all_core_metrics_are_in_core_bundle() {
+    List<Metric> coreMetrics = CoreMetrics.getMetrics();
+    List<String> missingMetricDefinitions = new ArrayList<>();
+    for (Metric metric : coreMetrics) {
+      if (metric.isHidden()) {
+        continue;
```
Comment (julienlancelot) [inline sonar-core/src/test/java/org/sonar/core/i18n/DefaultI18nTest.java]: You could check that hidden metrics have no bundle

File: sonar-core/src/main/resources/org/sonar/l10n/core.properties (hunk 2 above)
Comment (julienlancelot) [inline sonar-core/src/main/resources/org/sonar/l10n/core.properties]: Forget, as the metric is hidden I think no bundle should be added for this metric

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java [hunk 5]
```diff
@@ -2260,6 +2260,7 @@
    * @since 4.5
    */
   public static final Metric<String> DEVELOPMENT_COST = new Metric.Builder(DEVELOPMENT_COST_KEY, "SQALE Development Cost", Metric.ValueType.STRING)
+    .setDescription("SQALE development cost")
```
Comment (julienlancelot) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: Why adding the description here if it's set in the bundle ?

File: sonar-core/src/main/resources/org/sonar/l10n/core.properties [hunk 6]
```diff
@@ -1341,15 +1341,7 @@ widget.treemap-widget.property.maxItems.desc=Maximum number of components to sho
...
-                      <li>Or simply browse the <a href="http://redirect.sonarsource.com/3">complete documentation</a>?</li>\
-                      <li>If you have a question or an issue, please visit the <a href="http://www.sonarsource.org/get-support/">Get Support</a> page.</li>\
-                    </ul>
+widget.welcome.html=<h3 class="marginbottom5">Welcome to SonarQube Dashboard</h3><p class="marginbottom5">Since you are able to read this, it means that you have successfully started your SonarQube server. Well done!</p><p class="marginbottom5">If you have not removed this text, it also means that you have not yet played much with SonarQube. So here are a few pointers for your next step:</p> <ul class="bullet"><li>Do you now want to <a href="http://redirect.sonarsource.com/1">run analysis</a> on a project?</li> <li>Maybe start <a href="http://redirect.sonarsource.com/2">customizing dashboards</a>?</li><li>Or simply browse the <a href="http://redirect.sonarsource.com/3">complete documentation</a>?</li><li>If you have a question or an issue, please visit the <a href="http://www.sonarsource.org/get-support/">Get Support</a> page.</li></ul>
```
Comment (julienlancelot) [inline sonar-core/src/main/resources/org/sonar/l10n/core.properties]: I don't see the link between the comment commit and the change.

File: server/sonar-server/src/main/java/org/sonar/server/metric/ws/SearchAction.java [hunk 7]
```diff
@@ -83,7 +82,7 @@ public void handle(Request request, Response response) throws Exception {
     Boolean isCustom = request.paramAsBoolean(PARAM_IS_CUSTOM);
     DbSession dbSession = dbClient.openSession(false);
     try {
-      List<MetricDto> metrics = dbClient.metricDao().selectEnabled(dbSession, isCustom, searchOptions.getOffset(), searchOptions.getLimit());
+      List<MetricDto> metrics = dbClient.metricDao().selectEnabledSortedByKey(dbSession, isCustom, searchOptions.getOffset(), searchOptions.getLimit());
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/metric/ws/SearchAction.java]: I know that we've talked about this, but in fact you should keep the sort by name from DB : every metrics should have a name in DB, but they have not necessarily a name in the bundle.
Let's talk about that.

**Comment by julienlancelot** [comment]
WS should not load metrics name/description from english bundle, but form database (as it's the same value).

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java (hunk 5 above)
Comment (teryk) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: I think we want both systematically, no?

Comment (julienlancelot) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: Right !

**Comment by julienlancelot** [comment]
Please merge last 2 commits and LGTM

**Pull request closed without merging**